### ISSUE TITLE
Add an `exists` keyword as sugar on `collect`

### ISSUE THREAD

**Issue by keenancrane** [penrose/penrose#1469]
Thanks to @liangyiliang we now have a `collect` keyword that collects a set of matches into a list.

This keyword can be used to achieve functionality equivalent to `exists` (as described in #290), via

```
collect Type t into ts {
   -- this rule will be invoked exactly once if there is any instance of Type
}
```

In the case where one wants to emulate `exists` functionality, the list `ts` effectively serves as a dummy variable (i.e., it need not be used).  However, the semantics would be much clearer if we could simply write

```
exists Type t {
   -- do something just once
}
```

> **Proposal:** Make `exists Type t` sugar for `collect Type t into ts`.

Not sure how hard that is on the implementation front…

**Comment by liangyiliang** [comment]
I don't think that would be difficult to implement. The semantics differ by a little bit: we do need to make sure that within the `exists` block, `ts` is not an accessible variable.

**Comment by samestep** [comment]
Isn't this a duplicate of #290?

**Comment by liangyiliang** [comment]
> Isn't this a duplicate of #290?

Yes, with one key difference.

Previously we didn't have `collection`, so `exists` has nothing to rely on. Now we do have `collection`, `exists` is nothing but a syntactic sugar for `collection`.

**Comment by keenancrane** [comment]
Yeah, as @liangyiliang says this is an implementation path specifically for `foreach` using `collection`, whereas `exists` is just one of several language issues/questions discussed in #290 (permutations vs. combinations, distinctness and identity, composing the features, lists).

**Comment by keenancrane** [comment]
Some details/questions that we should consider when finalizing the design of `exists`: https://github.com/penrose/penrose/issues/290#issuecomment-1454174172
